src/modules/client/components/clientAlerts/getClientAlerts.tsx

```diff
@@ -0,0 +1,37 @@
+import { ClientAlertProps } from '@/modules/client/components/clientAlerts/ClientAlert';
+import { clientBriefName } from '@/modules/hmis/hmisUtil';
+import { ClientWithAlertFieldsFragment } from '@/types/gqlTypes';
+
+enum AlertPriority {
+  high = 3,
+  medium = 2,
+  low = 1,
+}
+
+export function getClientAlerts(
```

Comment:
I like that this is factored out, but I would probably make it a hook and memoize the alerts so they are not re-calculated on every render.

```
function useClientAlerts(clients, shouldShowClientName) {

    const clientAlerts = useMemo(() => {
          // code to construct and sort the Client Alerts

     }, [clients, shouldShowClientName]}) // only re-run when these change
    

   return { clientAlerts }
}
```

Reply:
Another way to do this would be to make the hook responsible for fetching the data, if any. For example:

`useClientAlerts ` hook accepts arguments `{ householdId?: string, client?: ClientWithAlertFieldsFragment, showClientName?: boolean }`
The hook would be responsible for:
1. Fetch alerts for household if `householdId` was provided (use `skip` to skip query otherwise)
2. If the `client` was provided, use the alerts on it
3. Return a properly sorted array with type `ClientAlertFieldsFragment[]`

That type of conditional-fetching-hook is okay, I think, it makes the component code a lot cleaner. The `useFormDefinition` is similar.

Let me know if you want to chat/pair through these ideas

Reply:
@gigxz thank you for the suggestions! I like the idea of the cleaner code that this could achieve but I wasn't quite able to get it working -- see https://github.com/greenriver/hmis-frontend/compare/martha/client-alerts...martha/client-alert-hooks for my attempts, but yes I would love to pair or chat more to get a better sense of how to put this together.

Reply:
Ok great! Lets. 
This one is annoying, you just need to cheat to make typescript happy:
> variables: { id: householdId || '' },
>     skip: !householdId,